store/daystore/day_store.go

```diff
@@ -124,11 +127,20 @@ func (ds *DayStore) Put(tp statstore.TimePoint) error {
 		max:         max,
 		ninetyFifth: nf,
 	}
-	if ds.size < 24 {
-		ds.size += 1
+
+	// check if the TimePoint is multiple hours in the future
+	// insert the hourEntry the appropriate amount of hours
+	distance := tp.Timestamp.Sub(ds.lastFlush)
+	nextflush := tp.Timestamp
+	for distance.Nanoseconds() >= time.Hour.Nanoseconds() {
```

Comment:
I assume this logic still requires input in a chronological order. Make sure you document that.